pkg/kev/config/extensions_test.go

```diff
@@ -105,38 +101,55 @@ var _ = Describe("Extentions", func() {
 					}
 				})
 
+				JustBeforeEach(func() {
+					svc.Extensions = extensions
+				})
+
 				When("workload is invalid", func() {
-					It("returns an error", func() {
-						parsedCfg, err = config.K8SCfgFromMap(extensions)
-						Expect(err).To(HaveOccurred())
-						Expect(err.Error()).To(ContainSubstring("Workload.Replicas is required"))
+					It("it is ignored and returns defaults", func() {
+						parsedCfg, err = config.K8SCfgFromCompose(&svc)
+						Expect(err).NotTo(HaveOccurred())
 					})
 				})
 			})
 
 			Context("missing liveness probe type", func() {
 				BeforeEach(func() {
+					k8s.Workload.Type = config.DefaultWorkload
 					k8s.Workload.Replicas = 10
 				})
 
 				It("returns error", func() {
-					parsedCfg, err = config.K8SCfgFromMap(extensions)
-					Expect(err).To(HaveOccurred())
-					Expect(err.Error()).To(Equal("Workload.LivenessProbe.Type is required"))
+					parsedCfg, err = config.K8SCfgFromCompose(&svc)
+					Expect(err).NotTo(HaveOccurred())
 				})
 			})
 
 			Context("missing replicas", func() {
 				BeforeEach(func() {
+					k8s.Workload.Type = config.DefaultWorkload
 					k8s.Workload.LivenessProbe.Type = config.ProbeTypeNone.String()
 				})
 
-				It("returns error", func() {
-					parsedCfg, err = config.K8SCfgFromMap(extensions)
-					Expect(err).To(HaveOccurred())
-					Expect(err.Error()).To(Equal("Workload.Replicas is required"))
+				It("returns in defaults", func() {
+					parsedCfg, err = config.K8SCfgFromCompose(&svc)
+					Expect(err).NotTo(HaveOccurred())
```

Comment:
I think we're missing the assertion here that verifies the returned value is in fact a default value?

Reply:
fixed

Reply:
it'd be good to validate whether `parsedCfg` is as expected though.